hazelcast-client/src/main/java/com/hazelcast/client/proxy/ClientRingbufferProxy.java

```diff
@@ -0,0 +1,167 @@
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.proxy;
+
+import com.hazelcast.client.spi.ClientProxy;
+import com.hazelcast.client.spi.impl.ClientInvocation;
+import com.hazelcast.client.spi.impl.ClientInvocationFuture;
+import com.hazelcast.core.ICompletableFuture;
+import com.hazelcast.core.IFunction;
+import com.hazelcast.nio.serialization.Data;
+import com.hazelcast.ringbuffer.OverflowPolicy;
+import com.hazelcast.ringbuffer.ReadResultSet;
+import com.hazelcast.ringbuffer.Ringbuffer;
+import com.hazelcast.ringbuffer.impl.client.AddAllRequest;
+import com.hazelcast.ringbuffer.impl.client.AddAsyncRequest;
+import com.hazelcast.ringbuffer.impl.client.AddRequest;
+import com.hazelcast.ringbuffer.impl.client.CapacityRequest;
+import com.hazelcast.ringbuffer.impl.client.HeadSequenceRequest;
+import com.hazelcast.ringbuffer.impl.client.ReadManyRequest;
+import com.hazelcast.ringbuffer.impl.client.ReadOneRequest;
+import com.hazelcast.ringbuffer.impl.client.RemainingCapacityRequest;
+import com.hazelcast.ringbuffer.impl.client.SizeRequest;
+import com.hazelcast.ringbuffer.impl.client.TailSequenceRequest;
+
+import java.util.Collection;
+
+import static com.hazelcast.ringbuffer.impl.RingbufferProxy.MAX_BATCH_SIZE;
+import static com.hazelcast.util.Preconditions.checkFalse;
+import static com.hazelcast.util.Preconditions.checkNotNegative;
+import static com.hazelcast.util.Preconditions.checkNotNull;
+import static com.hazelcast.util.Preconditions.checkTrue;
+
+public class ClientRingbufferProxy<E> extends ClientProxy implements Ringbuffer<E> {
+
+    private volatile long capacity = -1;
+
+    public ClientRingbufferProxy(String serviceName, String objectName) {
+        super(serviceName, objectName);
+    }
+
+    @Override
+    public long capacity() {
+        if (capacity == -1) {
+            CapacityRequest request = new CapacityRequest(getName());
+            capacity = invoke(request);
+        }
+
+        return capacity;
+    }
+
+    @Override
+    public long size() {
+        SizeRequest request = new SizeRequest(getName());
+        Long result = invoke(request);
+        return result;
+    }
+
+    @Override
+    public long tailSequence() {
+        TailSequenceRequest request = new TailSequenceRequest(getName());
+        Long result = invoke(request);
+        return result;
+    }
+
+    @Override
+    public long headSequence() {
+        HeadSequenceRequest request = new HeadSequenceRequest(getName());
+        Long result = invoke(request);
+        return result;
+    }
+
+    @Override
+    public long remainingCapacity() {
+        RemainingCapacityRequest request = new RemainingCapacityRequest(getName());
+        Long result = invoke(request);
+        return result;
+    }
+
+    @Override
+    public long add(E item) {
+        checkNotNull(item, "item can't be null");
+
+        AddRequest request = new AddRequest(getName(), toData(item));
+        Long result = invoke(request);
```

Comment:
Should these requests to be send to node that owning a specific partition ? 
If yes, these seem to be send to random node. It is similar problem in #5861. 
see https://github.com/hazelcast/hazelcast/pull/5861/files#r36734618